datalad/plugin/addurls.py

```diff
@@ -556,30 +556,24 @@ def add_meta(rows):
     """
     from unittest.mock import patch
 
+    # OPT: group by dataset first so to not patch/unpatch always_commit
+    # per each file of which we could have thousands
+    from collections import defaultdict
+    dss_rows = defaultdict(list)
     for row in rows:
-        ds, filename = row["ds"], row["ds_filename"]
+        dss_rows[row["ds"]].append(row)
+
+    for ds, ds_rows in dss_rows.items():
```

Comment:
Again, fine as is, but in case you didn't consider it: `itertools.groupby` would work nicely here.

Reply:
Great idea, will use it and that will remove aforementioned import.